eppo_core/src/event_ingestion/context.rs

```diff
@@ -0,0 +1,73 @@
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+use crate::Str;
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, derive_more::From)]
+#[serde(untagged)]
+pub enum ContextValue {
+    String(Str),
+    Number(f64),
+    Boolean(bool),
+    Null,
```

Comment:
minor: I think we want to model a potentially absent value as `Option<ContextValue>`. This will make sure that we don't send `null` fields and instead erase fields.

This also works well with deserialization because `Option<>` parses `null`, so we can get rid of `try_from_json()` and `ContextError`